### PR TITLE
Document that _repr_*_() can return metadata

### PR DESCRIPTION
**Pull request by zichongkao**
This should address this open issue: https://github.com/ipython/ipython/issues/10973.

I didn't change the api docs because they already seem to have documentation for returning metadata.

### CODE REVIEW

File: docs/source/config/integrating.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ Rich display
 ============
 
 The notebook and the Qt console can display richer representations of objects.
-To use this, you can define any of a number of ``_repr_*_()`` methods. Note that
```
Comment (takluyver) [inline docs/source/config/integrating.rst]: This wording was actually what it was meant to say.

File: docs/source/config/integrating.rst [hunk 2]
```diff
@@ -42,6 +42,13 @@ For example::
         def _repr_html_(self):
             return "<h1>" + self.text + "</h1>"
 
+We often want to provide frontends with guidance on how to display the data. To
+support this, ``_repr_*_()`` methods can also return a data, metadata tuple where
```
Comment (takluyver) [inline docs/source/config/integrating.rst]: Let's put brackets around `(data, metadata)` for clarity.

File: docs/source/config/integrating.rst [hunk 3]
```diff
@@ -42,6 +42,13 @@ For example::
...
+metadata is a dictionary containing arbitrary key, value pairs for the frontend
```
Comment (takluyver) [inline docs/source/config/integrating.rst]: I think the typical way to write this is "key-value pairs"

File: docs/source/config/integrating.rst [hunk 4]
```diff
@@ -42,6 +42,13 @@ For example::
...
+to interpret. An example use case is ``_repr_jpeg()``, which can be set to
```
Comment (takluyver) [inline docs/source/config/integrating.rst]: `_repr_jpeg_()` - you missed an underscore.

**Comment by fperez** [comment]
Thanks for your contribution, @zichongkao! 

@takluyver, I see the CI checks are failing, but this is a doc-only PR. Are they generically busted? Sorry that I haven't been keeping an eye on this for so long, so feel free to tell me if it's documented elsewhere or you're working on it :)

**Review by fperez — CHANGES_REQUESTED** [review]
Thanks for this! With a few small tweaks, should be ready for merging.

**Comment by takluyver** [comment]
Hi @fperez ! I'm glad you still get time to look at the code from time to time. :-)

Some change in Jedi broke one of our tests - it's tracked as #11103, and @Carreau was investigating, but possibly someone else needs to take over on that.

**Comment by zichongkao** [comment]
@fperez @takluyver Good points! Thanks for the feedback!

**Comment by fperez** [comment]
Merged, thanks @zichongkao for your first contribution to IPython! 

Looking forward to more in the future :)

**Pull request merged**